### PR TITLE
New voting

### PR DESCRIPTION
**Pull request by bbmcmann**
OK, so this is most of the implementation for voting. 
What I changed: 
-review schema and rout functionality to reflect voting arrays of user id's instead of vote counts
-added logic to check if user has voted on a review on render
-change review doc when a user votes by adding and removing them from the correct lists
-added styling for voting indication
-fixed rendering errors
-changed review data  formatting to match the db
Right now, it is set up to use a test ID representing the current logged in user. I will need to use the user cookies from a future merge to get this to work for the signed in user.

**_To test:_**
go to the Cars movie page. One review (should be the first) should have some upvotes and downvotes, and one of the vote arrows will be large and yellow. This indicates the user's selected vote. You can click the other arrow to see the arrow switch along with the numbers. You can see this in the DB as well by filtering reviews by movie_id: 920. it should also be open of the first reviews, you can look at the arrays to see them change. You can also console.log it to see the differences

### CODE REVIEW

**Comment by bbmcmann** [comment]
its gonna fail the review tests, which i will fix when i add the users in

File: frontend/src/components/Review.jsx [hunk 1]
```diff
@@ -54,36 +55,88 @@ const VoteDiv = styled.div`
...
     if (!voted) {
       setVoted(true);
+      console.log("voted");
       setVote(thisVote);
       if (thisVote === "up") {
         setCurUpVote(curUpVote + 1);
       } else {
         setCurDownVote(curDownVote + 1);
       }
+      if (thisVote === "up") {
```
Comment (jlaksana) [inline frontend/src/components/Review.jsx:135]: You already have an if else above this with "up" or "down" so you can remove this one and add the lines to the above statements. Same for lines 149-153

File: frontend/src/components/Review.jsx [hunk 2]
```diff
@@ -54,36 +55,88 @@ const VoteDiv = styled.div`
...
+      new_downvotes = downvote_list.filter((id) => id !== logged_in_user);
+      upvote_list.push(logged_in_user);
+      new_upvotes = upvote_list;
+    } else {
+      new_upvotes = upvote_list.filter((id) => id !== logged_in_user);
+      downvote_list.push(logged_in_user);
+      new_downvotes = downvote_list;
+    }
+    axios
+      .patch(`http://localhost:5000/reviews/${_id}`, {
```
Comment (jlaksana) [inline frontend/src/components/Review.jsx:117]: We should start using the backend url env variable for our deployment

File: frontend/src/components/Review.jsx (hunk 1 above)
Comment (bbmcmann) [reply frontend/src/components/Review.jsx:135]: oh yep good catch, didn't even notice while I was shifting blocks around haha

**Comment by jlaksana** [comment]
Two comments above. Otherwise looks great. Good work and thanks for taking this on!

**Review by pullingrm12 — COMMENTED** [review]
I was able to go to cars to test the voting system, and I can see the highlighted arrow for the current vote. I can switch between upvote and downvote and I can't do more than one. 

The code looks good outside of the comments Jonathan made, I will say the database information for the upvote / downvote differ from the values shown on the movie page but that isn't a big issue at the moment. 

For the cookies, you can look at the "link_users_to_front" branch and look at "profile.jsx" to see how you're able to use the stored cookies if you need to. 

Great job on this, looks good!

**Comment by dh435** [comment]
Looks great! 
The database is updating correctly to reflect the change in upvotes/downvotes, 
Aside from the unsuccessful CI tests, it looks ready to merge with the main.

Thanks, for getting this stuff up and running!

**Pull request merged**